packages/nextra-theme-blog/src/utils/get-tags.ts

```diff
@@ -1,9 +1,33 @@
-import { PageMapItem } from 'nextra'
+import { PageMapItem, PageOpts } from 'nextra'
 
 export function split(tags: string | string[]): string[] {
   return (Array.isArray(tags) ? tags : tags.split(',')).map(s => s.trim())
 }
 
+const flattenPageMap = (page: PageMapItem, result: PageMapItem[] = []) => {
+  if (Array.isArray(page.children)) {
+    page.children.forEach(p => flattenPageMap(p, result))
+  }
+  result.push(page)
+}
+
+const flattenPageMaps = (pages: PageMapItem[], result: PageMapItem[] = []) => {
+  pages.forEach(v => flattenPageMap(v, result))
+}
```

Comment:
follow function is more readable
https://github.com/shuding/nextra/blob/4157b71f1dd4d0708960c2b8c26e63960f01599d/packages/nextra-theme-docs/src/utils/flatten.ts#L1-L7

Reply:
I don't think so. two `forEach` is much simpler then `reduce`. I am happy with current implementation.

Reply:
your implementation mutates `result` variable, the above implementation not

Reply:
I dont want to argue. I am here beacause i want to contribute. You are free to open another pr to `fix` this.